### PR TITLE
fix(GC): check refcount map fix

### PR DESCRIPTION
**Pull request by Kouprin**
Some blocks may have height higher than `head.height`. They must be counted too.

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -3352,7 +3352,7 @@ pub fn collect_receipts_from_response(
 pub fn check_refcount_map(chain: &mut Chain) -> Result<(), Error> {
     let head = chain.head()?;
     let mut block_refcounts = HashMap::new();
-    for height in chain.store().get_genesis_height() + 1..=head.height {
+    for height in chain.store().get_genesis_height() + 1..=head.height + 100 {
```
Comment (bowenwang1996) [inline chain/chain/src/chain.rs]: Where does 100 come from?

Comment (Kouprin) [reply chain/chain/src/chain.rs]: Usually we don't create tests with more than 200-300 blocks. I guess, + 100 should cover any sanity case.

Comment (bowenwang1996) [reply chain/chain/src/chain.rs]: I don't understand. Why do you need to go past head height at all?

Comment (Kouprin) [reply chain/chain/src/chain.rs]: Screen Shot 2020-03-31 at 7 30 37 PM
Running tests/adversarial/start_from_genesis.py overtake doomslug_off, test0_finished/stdout:
Screen Shot 2020-03-31 at 7 31 08 PM
I suppose it's related to the idea that we can receive blocks which are ahead of current head or even build a separate chain. It seems `head.height` is a bad indicator of what height is the latest block in the storage. `latest_known` doesn't work this way too.

Comment (Kouprin) [reply chain/chain/src/chain.rs:3361]: Could you suggest another solution for failing Nightly tests?

Comment (bowenwang1996) [reply chain/chain/src/chain.rs:3361]: Which one failed because of this?

Comment (Kouprin) [reply chain/chain/src/chain.rs:3361]: Some of them. Please check Nightly failures.
This one may also help: https://near-protocol.slack.com/archives/CR4NE2MCK/p1585657931003400

Comment (Kouprin) [reply chain/chain/src/chain.rs:3361]: @bowenwang1996 any progress here?
It's possible to apply some workarounds to ignore blocks with heights higher than `head.height`, however I not agree with this way of problem solving. Why we should not check a part of Reference Map where heights are higher than `head.height`?

Comment (bowenwang1996) [reply chain/chain/src/chain.rs:3361]: I believe this problem will cease to exist when we have the new doomslug where block with highest height is the head.

Comment (Kouprin) [reply chain/chain/src/chain.rs:3361]: Is there an issue related to the Doomslug update? Could you create it if not?

Comment (bowenwang1996) [reply chain/chain/src/chain.rs:3361]: https://github.com/nearprotocol/nearcore/issues/2342

Comment (Kouprin) [reply chain/chain/src/chain.rs]: @bowenwang1996 I've updated this PR
Before #2342 we will process +100 heights in refmap to check that refmap is calculating properly. After #2342 we will remove unnecessary validations.

**Pull request merged**